src/test/java/com/vinyl/AddProductToCartTest.java

```diff
@@ -0,0 +1,132 @@
+package com.vinyl;
+
+import com.vinyl.helper.TokenHeaderHelper;
+import com.vinyl.model.Cart;
+import com.vinyl.model.Product;
+import com.vinyl.model.ProductCart;
+import com.vinyl.model.Token;
+import com.vinyl.modelDTO.AddProductToCartDTO;
+import com.vinyl.repository.CartRepository;
+import com.vinyl.repository.ProductCartRepository;
+import com.vinyl.repository.TokenRepository;
+import org.assertj.core.api.Assertions;
+import org.assertj.core.util.Lists;
+import org.junit.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.*;
+
+import java.time.LocalDate;
+
+public class AddProductToCartTest extends BaseIntegration {
+
+    @Autowired
+    private TokenHeaderHelper tokenHeaderHelper;
+
+    @Autowired
+    private ProductCartRepository productCartRepository;
+
+    @Autowired
+    private CartRepository cartRepository;
+
+    @Autowired
+    private TokenRepository tokenRepository;
+
+    private AddProductToCartDTO request;
+
+    private Product product;
+
+    @Override
+    public void setUp() {
+        super.setUp();
+        product = defaultEntitiesHelper.createProduct();
+        request = new AddProductToCartDTO();
+        request.setQuantity(3);
+    }
+
+    public ResponseEntity<?> setUpHeaderAndGetTheResponse(Token token){
```

Comment:
move this after the tests, and make it private (I assume it is only used in this class)

Reply:
fixed & done the same thing for GetCartDetailsTest class